### PR TITLE
test: add StartWorkAction edge case tests

### PR DESCRIPTION
**Pull request by mthalman**
test: add StartWorkAction edge case tests

Add 8 edge-case tests to StartWorkAction covering:

- Worktree failure with branch rollback also failing
- Fallback to origin/main when origin/dev does not exist
- Fallback to HEAD when neither origin/dev nor origin/main exist
- User cancels repo selection in multi-workspace scenario
- No git repository found in workspace folders
- Undefined externalId handling
- Git branch creation command failure
- Worktree directory already exists (cleanup verification)

### CODE REVIEW

File: packages/github/src/test/startWorkAction.test.ts [hunk 1]
```diff
@@ -269,5 +269,160 @@ describe('StartWorkAction', () => {
...
+      let worktreeSeen = false;
+      vi.mocked(execFile).mockImplementation(((cmd: string, args: string[], opts: any, cb: Function) => {
+        if (args[0] === 'worktree') {
+          worktreeSeen = true;
+          cb(new Error('worktree add failed'), '', '');
+        } else if (worktreeSeen && args[0] === 'branch' && args[1] === '-D') {
+          cb(new Error('branch delete failed'), '', '');
+        } else {
+          cb(null, { stdout: '', stderr: '' }, '');
+        }
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/test/startWorkAction.test.ts:283]: This mock callback pattern matches the existing beforeEach mock at line 50 established throughout this test file. It works correctly with promisify because without __promisify__, util.promisify resolves with the first non-error callback argument, which is the { stdout, stderr } object. The code then destructures { stdout } from this successfully. Changing only the new tests would create inconsistency with the existing pattern.

File: packages/github/src/test/startWorkAction.test.ts [hunk 2]
```diff
@@ -269,5 +269,160 @@ describe('StartWorkAction', () => {
...
+        expect.stringContaining('fatal: not a valid object name'),
+      );
+    });
+
+    it('handles worktree directory check when branch was already created', async () => {
+      let branchCreated = false;
+      vi.mocked(execFile).mockImplementation(((cmd: string, args: string[], opts: any, cb: Function) => {
+        if (args[0] === 'branch' && args[1] !== '--list' && args[1] !== '-D') {
+          branchCreated = true;
+        }
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/test/startWorkAction.test.ts]: Fixed: removed the unused branchCreated variable and simplified the mock. The test still verifies cleanup behavior through the deleteCalls assertion.
